refactor(napi/oxlint): rename `rules` to `ruleNames`

overlookmotel opened the pull request:
Pure refactor - just renaming. Rename `rules` to `rule_names`, because that's what it is!

camc314 commented:
### Merge activity

* **Jul 23, 2:18 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(napi/oxlint): rename `rules` to `ruleNames` (#12477)

Pure refactor - just renaming. Rename `rules` to `rule_names`, because that's what it is!
